File: Python/StableMatching/myObjects.py
from typing import Dict, List

class Candidate:
    def __init__(self, name: str, preferredHospitals: Dict[str,int]):
        self.name: str = name
        self.preferredHospitals: Dict[str,int] = preferredHospitals

# ASSUMPTION: candidates are ordered by priority
class Hospital:
    def __init__(self, name: str, preferredCandidates: List[str]):
        self.name: str = name
        self.preferredCandidates: List[str] = preferredCandidates

File: JavaScript/StableMatching/myObjects.ts
export class Hospital {
    constructor(name: string, preferredCandidates: string[]) {
        this.name = name;
        this.preferredCandidates = preferredCandidates;
    }
    name: string ;
    preferredCandidates: string[];
}

export class Candidate {
    constructor(name: string, preferredHospitals: {[key: string]: number}) {
        this.name = name;
        this.preferredHospitals = preferredHospitals;
    }

    name: string = '';
    preferredHospitals: {[key: string]: number} = {}
}